modules/claims_api/app/workers/claims_api/report_unsuccessful_submissions.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+require 'sidekiq'
+
+module ClaimsApi
+  class ReportUnsuccessfulSubmissions
+    include Sidekiq::Worker
+
+    def perform
+      if Settings.claims_api.unsuccessful_report_enabled
+        @to = Time.zone.now
+        @from = @to.monday? ? 7.days.ago : 1.day.ago
+        @consumers = ClaimsApi::AutoEstablishedClaim.where(created_at: @from..@to).pluck(:source).uniq
+
+        ClaimsApi::UnsuccessfulReportMailer.build(totals, pending, errored, @from, @to).deliver_now
+      end
+    end
+
+    def errored
+      ClaimsApi::AutoEstablishedClaim.where(
+        created_at: @from..@to,
+        status: %w[errored]
+      ).order(:source, :status)
+    end
+
+    def pending
+      ClaimsApi::AutoEstablishedClaim.where(
+        created_at: @from..@to,
+        status: 'pending'
+      ).order(:source, :status)
+    end
+
+    def totals
+      @consumers.map do |name|
```

Comment:
nit: this logic won't scale all that well. i dont know how many consumers we have atm, and I know this is in the background, but any value in composing a single query to do this work?

Reply:
I agree but our consumer numbers have remained generally low at the moment, also a good candidate for a technical debt refactor in the future.